packages/frontend-2/components/projects/AddDialog.vue

```diff
@@ -39,31 +39,32 @@
                   disabled-item-tooltip="You dont have rights to create projects in this workspace"
                   class="flex-1"
                 />
-                <FormButton
-                  :icon-left="PlusIcon"
-                  hide-text
-                  class="flex"
-                  color="outline"
-                  @click="isCreatingWorkspace = true"
-                />
+                <div v-tippy="'Create workspace'" class="flex">
+                  <FormButton
+                    :icon-left="PlusIcon"
+                    hide-text
+                    class="flex"
+                    color="outline"
+                    @click="confirmCreateWorkspace"
+                  />
+                </div>
               </div>
             </div>
-            <FormButton v-else color="outline" @click="isCreatingWorkspace = true">
+            <FormButton v-else color="outline" @click="confirmCreateWorkspace">
               New workspace
             </FormButton>
             <p class="text-foreground-2 text-body-2xs">
-              Workspace offers better project management and higher data security.
+              Workspaces offer better project management and higher data security.
             </p>
           </div>
-          <ProjectsNewWorkspace
-            v-if="isCreatingWorkspace"
-            mixpanel-event-source="create-project-modal"
-            @cancel="isCreatingWorkspace = false"
-            @workspace-created="onWorkspaceCreated"
-          />
         </template>
       </div>
     </form>
+    <CommonConfirmDialog
+      v-model:open="showConfirmDialog"
+      text="You have unsaved changes. Are you sure you want to leave and create a new workspace?"
```

Comment:
Small thing, but maybe change the copy to `text="You have unsaved changes. Are you sure you want to leave?"` so we can also trigger it on cancel/close?

Reply:
I like this suggestion 👍